packages/sdk/src/util/webHost.ts

```diff
@@ -63,42 +68,47 @@ export class WebHost {
 	}
 
 	private serveStaticFiles(server: Restify.Server) {
-		server.get('/*',
-			// host static binaries
-			(req, res, next) => this.serveStaticBuffers(req, res, next),
-			// host static files
-			Restify.plugins.serveStatic({
-				directory: this._baseDir,
-				default: 'index.html'
-			}
-			));
+		server.get(`/buffers/:name`,
+			this.checkStaticBuffers,
+			Restify.plugins.conditionalRequest(),
+			this.serveStaticBuffers);
+		server.get('/*', Restify.plugins.serveStaticFiles(this._baseDir));
 	}
 
-	private readonly bufferRegex = new RegExp(`^/${BUFFER_KEYWORD}/(.+)$`, 'u');
-
-	private serveStaticBuffers(req: Restify.Request, res: Restify.Response, next: Restify.Next) {
-		// grab path part of URL
-		const matches = this.bufferRegex.exec(req.url);
-		const procPath = matches && matches[1] || null;
-
-		// see if there's a handler registered for it
-		if (!procPath || !this.bufferMap[procPath]) {
-			return next();
+	private checkStaticBuffers = (req: Restify.Request, res: Restify.Response, next: Restify.Next) => {
+		const info = this.bufferMap[req.params.name];
+		if (info) {
+			res.setHeader('ETag', info.etag);
+			next();
+		} else {
+			next(new Error(`No buffer registered under name ${req.params.name}`));
```

Comment:
What happens with this error? What result code is returned to the client in this case?

Reply:
I had thought this would return an http error code, but I'm not positive. Let me test it. Mostly just copied that pattern from a restify sample.

Reply:
It used to throw a 500 error. I've updated it to be a 404.